app/uk/gov/hmrc/gform/commons/HttpFunctions.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.commons
+
+import uk.gov.hmrc.http.HttpReads.{ is2xx, is4xx, is5xx }
+import uk.gov.hmrc.http._
+
+trait HttpFunctions {
+  def jsonHttpReads[A](rds: HttpReads[A]): HttpReads[A] =
```

Comment:
Why is this functionality implemented here? Shouldn't this be part of some platform-wide library?

Reply:
I checked the docs, the new implementation doesn't exactly replicate the old deprecated reader and expects us to implement custom one if we need the exact behaviour as before. The old implementation looks tailored to a specific use case. 

Personally, I would expect the framework to do minimal when it comes to response processing and let the application decide what to do

Reply:
But why we want to stick with an old implementation? This `HttpReads` this builds on seems for some legacy purposes. Why not to switch to non-legacy way of doing http calls?

Reply:
By that I mean whats wrong with using
```
import uk.gov.hmrc.http.HttpReads.Implicits._
```
instead of this implementation?

Reply:
We use BadRequestException and NotFoundException in other layers. Using the implicits means updating the handling of exceptions in the top layers. These exceptions are also being thrown in other areas of code.

Reply:
> Using the implicits means updating the handling of exceptions in the top layers

This is one place to update (perhaps couple of lines of code) vs using this wrapper anytime we want to make a call.

> These exceptions are also being thrown in other areas of code.

I think this is ok, no need to do anything about this. This is client duty to handle appropriately isn't it?

Reply:
Ok, will take a look